mbofui/app/js/controllers.js

```diff
@@ -1,112 +1,214 @@
-
-        function posSuccess(pos, map) {
-            $scope.coords = pos.coords;
-            // serialize the inputs to create a URL
-            var url = '/api/messages/';
-            //2016-02-22T23:39:29Z
-            var startTime = moment($scope.newStartTime).format('YYYY-MM-DDTHH:mm:ssZ');
-            var endTime = moment($scope.newEndTime).format('YYYY-MM-DDTHH:mm:ssZ');
-            var postingTime = moment().format('YYYY-MM-DDTHH:mm:ssZ')
-            var data = {
-                "messageText" : $scope.newMessageText,
-                "startTime" : startTime,
-                "endTime" : endTime,
-                "latitude" : $scope.coords.latitude,
-                "longitude" : $scope.coords.longitude,
-                "altitudeMeters" : $scope.coords.accuracy,
-                "postingTime"  : postingTime
-            }
-            Bof.PostBof(url,data).then(function(result) {
-                var marker = new google.maps.Marker({
-                    position: { lat: result.data.latitude, lng: result.data.longitude },
-                    map: window.map,
-                    zIndex: 10000
-                });
-
-                google.maps.event.addListener(marker, 'click', function() {
-                    var infowindow = new google.maps.InfoWindow({
-                        content: result.data.messageText
-                    });
-                    infowindow.open(window.map, marker);
-                });
-                $scope.totalBofs = $scope.totalBofs + 1;
-
-                var iconFile = 'http://maps.google.com/mapfiles/ms/icons/yellow-dot.png'; 
-                marker.setIcon(iconFile); 
-            });
+/* global angular, ocellus, $, L, resolveCategory, resolveIcon, moment */
+
+ocellus.controller('mapController', ['$scope', '$filter', '$timeout', '$log', 'leafletData', 'Bof', function($scope, $filter, $timeout, $log, leafletData, Bof) {
+  angular.extend($scope, {
+    center: {
+      zoom: 16,
+      autoDiscover: true
+    },
+    events: {},
+    controls: {
+      draw: {}
+    },
+    layers: {
+      baselayers: {
+        googleRoadmap: {
+          name: 'Google Streets',
+          layerType: 'ROADMAP',
+          type: 'google',
+          layerParams: {
+            showOnSelector: true
+          }
+        },
+        googleHybrid: {
+          name: 'Google Hybrid',
+          layerType: 'HYBRID',
+          type: 'google'
+        },
+        googleTerrain: {
+          name: 'Google Terrain',
+          layerType: 'TERRAIN',
+          type: 'google'
+        },
+
+        mapbox_light: {
+          name: 'Mapbox Light',
+          url: 'https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}',
+          type: 'xyz',
+          layerOptions: {
+            id: 'gsilver.pk8alhme',
+            accessToken: 'pk.eyJ1IjoiZ3NpbHZlciIsImEiOiJjaW1xYmxianowMGZsdXJra2FjbXhpYjE4In0.LL9yfFdOwvatCyCbxBDW_A',
```

Comment:
There's a lot of duplication of code from `mbofui/app/app.js`.  Can't this take advantage of that code?